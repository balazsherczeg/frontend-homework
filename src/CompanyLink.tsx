import { connect } from 'react-redux'
import { createStructuredSelector } from 'reselect'

import { ReduxState } from './types'
import { getIsCompanySelected } from './selectors'
import { setSelectedCompanyId } from './actions'
import { Company } from './types'

import DropdownCloser from './DropdownCloser';

type OwnProps = {
  company: Company,
}

type ReduxProps = {
  isCompanySelected: boolean,
}

type DispatchProps = {
  setSelectedCompanyId: (id: number) => void,
}

export const CompanyLink = ({
  company: { name, id },
  isCompanySelected,
  setSelectedCompanyId
}: OwnProps & ReduxProps & DispatchProps) => (
  <DropdownCloser>
    <div
      className={`CompanyLink ${isCompanySelected ? "CompanyLink--selected" : ""}`}
      role="button"
      onClick={() => setSelectedCompanyId(id)}
    >
      <span className="strong">{name}</span>
      {isCompanySelected && <i className="material-icons">done</i>}
    </div>
  </DropdownCloser>
)

export default connect(
  createStructuredSelector<ReduxState, OwnProps, ReduxProps>({
    isCompanySelected: getIsCompanySelected,
  }),
  { setSelectedCompanyId }
)(CompanyLink)
