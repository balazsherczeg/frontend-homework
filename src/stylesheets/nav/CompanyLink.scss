.CompanyLink {
  @include menu-item;
  @include state-feedback;

  height: 44px;
  justify-content: space-between;

  &--selected {
    color: $dark-green;

    &, &:active {
      background: rgba($green, .1);
    }

    &:hover, &:focus {
      background: rgba($green, .2);
    }
  }
}
