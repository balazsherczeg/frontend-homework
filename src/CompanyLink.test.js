import { shallow } from 'enzyme'
import { CompanyLink } from './CompanyLink'

describe('<CompanyLink />', () => {
  const setSelectedCompanyIdMock = jest.fn()

  const render = (props) => shallow(
    <CompanyLink
      setSelectedCompanyId={setSelectedCompanyIdMock}
      company={{
        id: 1,
        name: "Dummy company",
      }}
      {...props }
    />
  )

  it('renders selected', () => {
    expect(render({ isCompanySelected: true })).toMatchSnapshot()
  })

  it('renders not selected', () => {
    expect(render({ isCompanySelected: false })).toMatchSnapshot()
  })

  it('calls setSelectedCompanyId when clicked', () => {
    const component = render({ isCompanySelected: false })
    component.find('[role="button"]').simulate('click')
    expect(setSelectedCompanyIdMock).toHaveBeenCalled()
  })
})
