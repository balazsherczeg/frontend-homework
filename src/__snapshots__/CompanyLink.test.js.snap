// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CompanyLink /> renders not selected 1`] = `
<Connect(DropdownCloser)>
  <div
    className="CompanyLink "
    onClick={[Function]}
    role="button"
  >
    <span
      className="strong"
    >
      Dummy company
    </span>
  </div>
</Connect(DropdownCloser)>
`;

exports[`<CompanyLink /> renders selected 1`] = `
<Connect(DropdownCloser)>
  <div
    className="CompanyLink CompanyLink--selected"
    onClick={[Function]}
    role="button"
  >
    <span
      className="strong"
    >
      Dummy company
    </span>
    <i
      className="material-icons"
    >
      done
    </i>
  </div>
</Connect(DropdownCloser)>
`;
